packages/vite/src/client/overlay.ts

```diff
@@ -165,39 +165,42 @@ kbd {
 `
 
 // Error Template
-const template = h(
-  'div',
-  { class: 'backdrop', part: 'backdrop' },
-  h(
-    'div',
-    { class: 'window', part: 'window' },
-    h(
-      'pre',
-      { class: 'message', part: 'message' },
-      h('span', { class: 'plugin', part: 'plugin' }),
-      h('span', { class: 'message-body', part: 'message-body' }),
-    ),
-    h('pre', { class: 'file', part: 'file' }),
-    h('pre', { class: 'frame', part: 'frame' }),
-    h('pre', { class: 'stack', part: 'stack' }),
-    h(
-      'div',
-      { class: 'tip', part: 'tip' },
-      'Click outside, press ',
-      h('kbd', {}, 'Esc'),
-      ' key, or fix the code to dismiss.',
-      h('br'),
-      'You can also disable this overlay by setting ',
-      h('code', { part: 'config-option-name' }, 'server.hmr.overlay'),
-      ' to ',
-      h('code', { part: 'config-option-value' }, 'false'),
-      ' in ',
-      h('code', { part: 'config-file-name' }, hmrConfigName),
-      '.',
-    ),
-  ),
-  h('style', {}, templateStyle),
-)
+const template =
```

Comment:
maybe we could init the `template` in a lazy way and set `enableOverlay` to `false` if `document` isn't available? Then we don't need to check if `template` is defined inside `ErrorOverlay` as it will never be created if there isn't a `document`

Reply:
To do that, I think we need three states instead of two (`true`/`false`).
If we set `enableOverlay` to `false` if `document` isn't available, when a server error happens, an overlay will be shown (by the `/@vite/client` in window) and the log will be output (by the `/@vite/client` in worker).
https://github.com/vitejs/vite/blob/445c4f21583334edb37c7b32a1474903a0852b01/packages/vite/src/client/client.ts#L276-L282

Reply:
Don't we have a similar issue already when `enableOverlay` is false? Two errors will be then logged to the console?
Maybe we should instead only show something (overlay or console.error) when document is defined?

Reply:
Ah, you're right. When `enableOverlay=false`, there's be duplicated error logs.

> Maybe we should instead only show something (overlay or console.error) when document is defined?

This sound good to me. I'll push a change.